Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -1270,6 +1277,9 @@ class CombinatorialKalmanFilter {
           state.options.direction * stepper.direction(state.stepping), &surface,
           nullptr);
 
+      detail::setupLoopProtection(state, stepper, result.pathLimitReached,
+                                  logger());
```

Comment:
Why do we have to re-set this up again here? Does this set it up for the next branch?

Reply:
we start propagating back to the target here so in order to make this stable we set a path limit here and later check if it was exceeded